bundle/definition/schema.go

```diff
@@ -0,0 +1,121 @@
+	ExclusiveMaximum     *float64               `json:"exclusiveMaximum,omitempty" mapstructure:"exclusiveMaximum,omitempty"`
+	ExclusiveMinimum     *float64               `json:"exclusiveMinimum,omitempty" mapstructure:"exclusiveMinimum,omitempty"`
+	Format               string                 `json:"format,omitempty" mapstructure:"format,omitempty"`
+	If                   *Schema                `json:"if,omitempty" mapstructure:"if,omitempty"`
+	//Items can be a Schema or an Array of Schema :(
+	Items         interface{} `json:"items,omitempty" mapstructure:"items,omitempty"`
+	Maximum       *float64    `json:"maximum,omitempty" mapstructure:"maximum,omitempty"`
+	MaxLength     *float64    `json:"maxLength,omitempty" mapstructure:"maxLength,omitempty"`
+	MinItems      *float64    `json:"minItems,omitempty" mapstructure:"minItems,omitempty"`
+	MinLength     *float64    `json:"minLength,omitempty" mapstructure:"minLength,omitempty"`
+	MinProperties *float64    `json:"minProperties,omitempty" mapstructure:"minProperties,omitempty"`
+	Minimum       *float64    `json:"minimum,omitempty" mapstructure:"minimum,omitempty"`
+	MultipleOf    *float64    `json:"multipleOf,omitempty" mapstructure:"multipleOf,omitempty"`
+	Not           *Schema     `json:"not,omitempty" mapstructure:"not,omitempty"`
+	OneOf         *Schema     `json:"oneOf,omitempty" mapstructure:"oneOf,omitempty"`
+
+	PatternProperties map[string]*Schema `json:"patternProperties,omitempty" mapstructure:"patternProperties,omitempty"`
+
+	Properties    map[string]*Schema `json:"properties,omitempty" mapstructure:"properties,omitempty"`
+	PropertyNames *Schema            `json:"propertyNames,omitempty" mapstructure:"propertyNames,omitempty"`
+	ReadOnly      *bool              `json:"readOnly,omitempty" mapstructure:"readOnly,omitempty"`
+	Required      []string           `json:"required,omitempty" mapstructure:"required,omitempty"`
+	Then          *Schema            `json:"then,omitempty" mapstructure:"then,omitempty"`
+	Title         string             `json:"title,omitempty" mapstructure:"title,omitempty"`
+	Type          interface{}        `json:"type,omitempty" mapstructure:"type,omitempty"`
+	UniqueItems   *bool              `json:"uniqueItems,omitempty" mapstructure:"uniqueItems,omitempty"`
+	WriteOnly     *bool              `json:"writeOnly,omitempty" mapstructure:"writeOnly,omitempty"`
+}
+
+// SingleType returns true if a schema has a single type declaration
+func (s *Schema) singleType() bool {
+	_, ok := s.Type.(string)
+	return ok
+}
+
+// GetType will return the singular type for a given schema and a success boolean. If the
+// schema does not have a single type, it will return the false boolean and an error.
+func (s *Schema) GetType() (string, bool, error) {
+	if !s.singleType() {
+		return "", false, errors.New("this schema has multiple types")
+	}
+	typeString, ok := s.Type.(string)
```

Comment:
I wonder if the previous call to `singleType` could not be directly handled here, as the cast here is exactly the one made by `singleType` (feels like we are doing the same thing twice). Could it be simplified with something like this?
```go
func (s *Schema) GetType() (string, bool, error) {
    typeString, ok := s.Type.(string)
    if !ok {
        return "", false, errors.New("this schema has multiple types")
    }
    return typeString, ok, nil
}
```

Reply:
You're correct, when I un-exported the other method I should have just done this.